Frameworks/UIKit/UIImage.mm

```diff
@@ -1033,59 +1035,41 @@ - (void)drawInRect:(CGRect)pos {
 */
 - (void)drawInRect:(CGRect)pos blendMode:(CGBlendMode)mode alpha:(float)alpha {
     //  [BUG: Need to honor mode and alpha]
-    CGContextRef cur = UIGraphicsGetCurrentContext();
+    CGContextRef ctx = UIGraphicsGetCurrentContext();
+    CGImageRef img = getImage(self);
 
-    if (!getImage(self)) {
+    if (img == NULL) {
         TraceWarning(TAG, L"m_pImage = NULL!");
         return;
     }
 
-    CGContextSaveGState(cur);
-    CGContextSetBlendMode(cur, mode);
+    CGContextSaveGState(ctx);
+    CGContextSetBlendMode(ctx, mode);
 
-    if (alpha != 1.0) {
+    if (alpha != 1.0f) {
         TraceVerbose(TAG, L"Should draw with alpha");
     }
 
-    if (_scale == 0) {
+    if (_scale == 0.0f) {
         TraceWarning(TAG, L"Scale should be non-zero!");
         return;
     }
 
-    if (_imageInsets.left == 0 && _imageInsets.top == 0 && _imageInsets.right == 0 && _imageInsets.bottom == 0) {
-        CGRect srcRect;
-        CGRect destRect = pos;
+    // Destination rect taken from pos with origin transformed from TL to BL
+    CGRect dstRect;
+    dstRect.origin = { pos.origin.x, (static_cast<float>(CGBitmapContextGetHeight(ctx)) - pos.origin.y - pos.size.height) };
+    dstRect.size = pos.size;
 
-        srcRect.origin.x = 0;
-        srcRect.origin.y = 0;
-        srcRect.size.width = float(getImage(self)->Backing()->Width());
-        srcRect.size.height = float(getImage(self)->Backing()->Height());
+    CGImageBacking* imgBacking = img->Backing();
 
-        //  Invert srcRect
-        srcRect.origin.y = getImage(self)->Backing()->Height() - srcRect.origin.y;
-        srcRect.size.height = -srcRect.size.height;
-        CGContextDrawImageRect(cur, getImage(self), srcRect, destRect);
-    } else {
-        static const CGPatternCallbacks patCB = { 0, drawNinePatchCallBack, NULL };
-        insetInfo ii = { self, makeRect(0, 0, pos.size.width * _scale, pos.size.height * _scale) };
-
-        CGPatternRef pat = CGPatternCreate((void*)&ii,
-                                           makeRect(0, 0, pos.size.width * _scale, pos.size.height * _scale),
-                                           CGAffineTransformMakeScale(_scale, _scale),
-                                           10,
-                                           10,
-                                           kCGPatternTilingConstantSpacing,
-                                           true,
-                                           &patCB);
-
-        CGFloat alpha = 1.0f;
-        CGContextSetFillPattern(cur, pat, &alpha);
-        CGContextFillRect(cur, pos);
-
-        CGPatternRelease(pat);
-    }
+    CGRect srcRect;
+    srcRect.origin = { 0 };
+    srcRect.size = { static_cast<float>(imgBacking->Width()), static_cast<float>(imgBacking->Height()) };
 
-    CGContextRestoreGState(cur);
+    // Draw image and divide into patches if necessary
+    drawPatches(ctx, self, &srcRect, &dstRect);
```

Comment:
With the original scaling this is what the datepicker output looks like:
![originalscaling](https://cloud.githubusercontent.com/assets/17790501/16824885/5325ac12-4924-11e6-81d1-50dbcda1e7f1.png)

With the scale factor not applied to the dest rect this is what is displayed:
![reviseddrawrect](https://cloud.githubusercontent.com/assets/17790501/16824887/5872b0ca-4924-11e6-926b-96970768b9a8.png)

In the first case it seems as though the image is overdrawn or scaled twice or something similar. In the second one it fits within the bounds.

I will try to replicate the scenario you described sometime tomorrow but in this particular case, keeping the original scaling does not seem to be the way to go.